src/Core/Authorization/AuthorizationResolver.cs

```diff
@@ -438,61 +440,150 @@ public IEnumerable<string> GetAllowedExposedColumns(string entityName, string ro
     {
-        Dictionary<string, Claim> claimsInRequestContext = new();
+        Dictionary<string, List<Claim>> resolvedClaims = new();
         if (context is null)
         {
-            return claimsInRequestContext;
+            return resolvedClaims;
         }
 
         string clientRoleHeader = context.Request.Headers[CLIENT_ROLE_HEADER].ToString();
 
         // Iterate through all the identities to populate claims in request context.
         foreach (ClaimsIdentity identity in context.User.Identities)
         {
-
-            // Only add a role claim which represents the role context evaluated for the request,
-            // as this can be via the virtue of an identity added by DAB.
-            if (!claimsInRequestContext.ContainsKey(AuthenticationOptions.ROLE_CLAIM_TYPE) &&
-                identity.HasClaim(type: AuthenticationOptions.ROLE_CLAIM_TYPE, value: clientRoleHeader))
-            {
-                claimsInRequestContext.Add(AuthenticationOptions.ROLE_CLAIM_TYPE, new Claim(AuthenticationOptions.ROLE_CLAIM_TYPE, clientRoleHeader, ClaimValueTypes.String));
-            }
-
             // If identity is not authenticated, we don't honor any other claims present in this identity.
             if (!identity.IsAuthenticated)
             {
                 continue;
             }
 
+            // DAB will only resolve one 'roles' claim whose value matches the x-ms-api-role header value
+            // because DAB executes requests in the context of a single role. The `roles` claim
+            // resolved here can be forwarded to MSSQL's set-session-context. Modifying this behavior
+            // is a breaking change.
+            if (!resolvedClaims.ContainsKey(AuthenticationOptions.ROLE_CLAIM_TYPE) &&
+                identity.HasClaim(type: AuthenticationOptions.ROLE_CLAIM_TYPE, value: clientRoleHeader))
+            {
+                List<Claim> roleClaim = new()
+                {
+                    new Claim(type: AuthenticationOptions.ROLE_CLAIM_TYPE, value: clientRoleHeader, valueType: ClaimValueTypes.String)
+                };
+
+                resolvedClaims.Add(AuthenticationOptions.ROLE_CLAIM_TYPE, roleClaim);
+            }
+
+            // Process all remaining claims and consolidate duplicate claim types into a list of claims.
             foreach (Claim claim in identity.Claims)
             {
-                /*
-                 * An example claim would be of format:
-                 * claim.Type: "user_email"
-                 * claim.Value: "authz@microsoft.com"
-                 * claim.ValueType: "string"
-                 */
-                // At this point, only add non-role claims to the collection and only throw an exception for duplicate non-role claims.
-                if (!claim.Type.Equals(AuthenticationOptions.ROLE_CLAIM_TYPE) && !claimsInRequestContext.TryAdd(claim.Type, claim))
+                // 'roles' claim has already been processed.
```

Comment:
nit: Having a more detailed explanation (like you have in your PR description) would make the code easier to understand.

For my understanding: Since you have already created a list with a value for `roles` claims type in the previous code block, you are ignoring other values for that same claim type(Only the first encountered value is honored), right?

Reply:
correct. Making comment clearer:
```
         // Process all remaining claims adding all `Claim` objects with the same claimType (claim name)
            // into a list and storing that in resolvedClaims using the claimType as the key.
```